moveit_simple/src/moveit_simple.cpp

```diff
@@ -1206,14 +1243,26 @@ bool Robot::getFK(const std::vector<double> & joint_point,
                   Eigen::Affine3d &pose) const
 {
   virtual_robot_state_->setJointGroupPositions(joint_group_, joint_point);
-  const std::vector<std::string> link_names = joint_group_->getLinkModelNames();
-  const int vc =  (int) virtual_robot_state_->getVariableCount();
-  if ( joint_point.size() == vc)
+
+  if (ik_transforms_found_)
   {
-    pose = virtual_robot_state_->getFrameTransform(link_names.back());
+    Eigen::Affine3d base_to_tool0 = virtual_robot_state_->getFrameTransform(ik_tip_frame_);
+    pose = base_to_tool0 * tool0_to_tcp_.inverse();
     return true;
-  }else{
-    return false;
+  }
+  else
+  {
+    const std::vector<std::string> link_names = joint_group_->getLinkModelNames();
+    const int vc =  (int) virtual_robot_state_->getVariableCount();
+    if (joint_point.size() == vc)
+    {
+      pose = virtual_robot_state_->getFrameTransform(link_names.back());
```

Comment:
Would this bit of logic be needed if `ik_tip_frame` was set in the constructor?

Reply:
Nope.